group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetricsShard.java

```diff
@@ -140,11 +139,17 @@ public GroupCoordinatorMetricsShard(
         this.topicPartition = Objects.requireNonNull(topicPartition);
     }
 
-    public void incrementNumClassicGroups(ClassicGroupState state) {
-        AtomicLong counter = classicGroupGauges.get(state);
-        if (counter != null) {
-            counter.incrementAndGet();
-        }
+    /**
+     * @return An empty ClassicGroupGauges.
+     */
+    public static Map<ClassicGroupState, AtomicLong> createClassicGroupGauge() {
+        return Utils.mkMap(
+            Utils.mkEntry(ClassicGroupState.PREPARING_REBALANCE, new AtomicLong(0)),
+            Utils.mkEntry(ClassicGroupState.COMPLETING_REBALANCE, new AtomicLong(0)),
+            Utils.mkEntry(ClassicGroupState.STABLE, new AtomicLong(0)),
+            Utils.mkEntry(ClassicGroupState.DEAD, new AtomicLong(0)),
+            Utils.mkEntry(ClassicGroupState.EMPTY, new AtomicLong(0))
+        );
```

Comment:
In my opinion, we don't need to prepare it like this. We should rather be safe on the query side and ensure that we default to zero if the state does not exist.